internal/gitprovider/gitea/gitea.go

```diff
@@ -0,0 +1,267 @@
+
+	ListPullRequests(
+		ctx context.Context,
+		owner string,
+		repo string,
+		opts *gitea.ListPullRequestsOptions,
+	) ([]*gitea.PullRequest, *gitea.Response, error)
+
+	GetPullRequests(
+		ctx context.Context,
+		owner string,
+		repo string,
+		number int,
+	) (*gitea.PullRequest, *gitea.Response, error)
+
+	AddLabelsToIssue(
+		ctx context.Context,
+		owner string,
+		repo string,
+		number int,
+		labels []string,
+	) ([]*gitea.Label, *gitea.Response, error)
+}
+
+// provider is a Gitea implementation of gitprovider.Interface.
+type provider struct { // nolint: revive
+	owner  string
+	repo   string
+	client giteaClient
+}
+
+// NewProvider returns a Gitea-based implementation of gitprovider.Interface.
+func NewProvider(
+	repoURL string,
+	_ *gitprovider.Options,
+) (gitprovider.Interface, error) {
+
+	host, owner, repo, err := parseRepoURL(repoURL)
+	if err != nil {
+		return nil, err
+	}
+	client, err := gitea.NewClient(host)
+	if err != nil {
+		return nil, err
+	}
+
+	return &provider{
+		owner:  owner,
+		repo:   repo,
+		client: &giteaClientWrapper{client},
+	}, nil
+}
+
+type giteaClientWrapper struct {
+	client *gitea.Client
+}
+
+func (g giteaClientWrapper) CreatePullRequest(
+	_ context.Context,
+	owner string,
+	repo string,
+	opts *gitea.CreatePullRequestOption,
+) (*gitea.PullRequest, *gitea.Response, error) {
+	return g.client.CreatePullRequest(owner, repo, *opts)
+}
+
+func (g giteaClientWrapper) ListPullRequests(
+	_ context.Context,
+	owner string,
+	repo string,
+	opts *gitea.ListPullRequestsOptions,
+) ([]*gitea.PullRequest, *gitea.Response, error) {
+	return g.client.ListRepoPullRequests(owner, repo, *opts)
+}
+
+func (g giteaClientWrapper) GetPullRequests(
+	_ context.Context,
+	owner string,
+	repo string,
+	number int,
+) (*gitea.PullRequest, *gitea.Response, error) {
+	return g.client.GetPullRequest(owner, repo, int64(number))
+}
+
+func (g giteaClientWrapper) AddLabelsToIssue(
+	_ context.Context,
+	owner string,
+	repo string,
+	number int,
+	_ []string,
+) ([]*gitea.Label, *gitea.Response, error) {
+	return g.client.AddIssueLabels(owner, repo, int64(number), gitea.IssueLabelsOption{})
+}
+
+// CreatePullRequest implements gitprovider.Interface.
+func (p *provider) CreatePullRequest(
+	ctx context.Context,
+	opts *gitprovider.CreatePullRequestOpts,
+) (*gitprovider.PullRequest, error) {
+	if opts == nil {
+		opts = &gitprovider.CreatePullRequestOpts{}
+	}
+	giteaPR, _, err := p.client.CreatePullRequest(ctx,
+		p.owner,
+		p.repo,
+		&gitea.CreatePullRequestOption{
+			Title: opts.Title,
+			Head:  opts.Head,
+			Base:  opts.Base,
+			Body:  opts.Description,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if giteaPR == nil {
+		return nil, fmt.Errorf("unexpected nil pull request")
+	}
+	pr := convertGiteaPR(*giteaPR)
+	if len(opts.Labels) > 0 {
+		_, _, err = p.client.AddLabelsToIssue(ctx,
+			p.owner,
+			p.repo,
+			int(pr.Number),
+			opts.Labels,
+		)
+	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	if len(opts.Labels) > 0 {
		if _, _, err = p.client.AddLabelsToIssue(ctx,
			p.owner,
			p.repo,
			int(pr.Number),
			opts.Labels,
		); err != nil {
			return nil, err
		}
	}
```

Reply:
Oh i'm not confortable with this notation ^^ Thanks!